src/server.c

```diff
@@ -1344,11 +1344,20 @@ void checkChildrenDone(void) {
     }
 }
 
+size_t getMemoryUsageAndUpdatePeak(){
+    size_t zmalloc_used = zmalloc_used_memory();
```

Comment:
we can pass used memory from outside, actually, the overhead of `zmalloc_used_memory()` is pretty high.